app/models/measure_condition_permutations/calculator.rb

```diff
@@ -0,0 +1,110 @@
+module MeasureConditionPermutations
+  class Calculator
+    # There are 2 rules for calculating permutations
+    #
+    # When there are conditions with matching permutation_keys, then a single
+    # permutation group is provided with all permutations
+    #
+    # If there are no conditions with matching permutation key, then generate
+    # one group per condition code, with a separate permutation per condition
+    # within it
+
+    def initialize(measure)
```

Comment:
So I'm definitely struggling to reason about this class (and I'm not convinced that's because of anything you've done)

Reply:
Should be simpler now - split into two separate calculators